test/mocks/v1/non_template_tests_test.go

```diff
@@ -0,0 +1,194 @@
+		Expect(err).NotTo(HaveOccurred())
+		cfg, err = kubeutils.GetConfig("", "")
+		Expect(err).NotTo(HaveOccurred())
+
+		clientset, err = apiext.NewForConfig(cfg)
+		Expect(err).NotTo(HaveOccurred())
+		// SimpleMockResource Constructor
+		simpleMockResourceClientFactory := &factory.MemoryResourceClientFactory{
+			Cache: memory.NewInMemoryResourceCache(),
+		}
+
+		simpleMockResourceClient, err = NewSimpleMockResourceClient(ctx, simpleMockResourceClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		// MockResource Constructor
+		mockResourceClientFactory := &factory.KubeResourceClientFactory{
+			Crd:         MockResourceCrd,
+			Cfg:         cfg,
+			SharedCache: kuberc.NewKubeCache(context.TODO()),
+		}
+
+		err = helpers.AddAndRegisterCrd(ctx, MockResourceCrd, clientset)
+		Expect(err).NotTo(HaveOccurred())
+
+		mockResourceClient, err = NewMockResourceClient(ctx, mockResourceClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		// FakeResource Constructor
+		fakeResourceClientFactory := &factory.MemoryResourceClientFactory{
+			Cache: memory.NewInMemoryResourceCache(),
+		}
+
+		fakeResourceClient, err = NewFakeResourceClient(ctx, fakeResourceClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		// AnotherMockResource Constructor
+		anotherMockResourceClientFactory := &factory.KubeResourceClientFactory{
+			Crd:         AnotherMockResourceCrd,
+			Cfg:         cfg,
+			SharedCache: kuberc.NewKubeCache(context.TODO()),
+		}
+
+		err = helpers.AddAndRegisterCrd(ctx, AnotherMockResourceCrd, clientset)
+		Expect(err).NotTo(HaveOccurred())
+
+		anotherMockResourceClient, err = NewAnotherMockResourceClient(ctx, anotherMockResourceClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		// ClusterResource Constructor
+		clusterResourceClientFactory := &factory.KubeResourceClientFactory{
+			Crd:         ClusterResourceCrd,
+			Cfg:         cfg,
+			SharedCache: kuberc.NewKubeCache(context.TODO()),
+		}
+
+		err = helpers.AddAndRegisterCrd(ctx, ClusterResourceCrd, clientset)
+		Expect(err).NotTo(HaveOccurred())
+
+		clusterResourceClient, err = NewClusterResourceClient(ctx, clusterResourceClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		// MockCustomType Constructor
+		mockCustomTypeClientFactory := &factory.MemoryResourceClientFactory{
+			Cache: memory.NewInMemoryResourceCache(),
+		}
+
+		mockCustomTypeClient, err = NewMockCustomTypeClient(ctx, mockCustomTypeClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		// Pod Constructor
+		podClientFactory := &factory.MemoryResourceClientFactory{
+			Cache: memory.NewInMemoryResourceCache(),
+		}
+
+		podClient, err = github_com_solo_io_solo_kit_pkg_api_v1_resources_common_kubernetes.NewPodClient(ctx, podClientFactory)
+		Expect(err).NotTo(HaveOccurred())
+		emitter = NewTestingEmitter(simpleMockResourceClient, mockResourceClient, fakeResourceClient, anotherMockResourceClient, clusterResourceClient, mockCustomTypeClient, podClient)
```

Comment:
nit: We have somewhat confusing mocks, but there is a v1 package (this one) with many clients, and a v1alpha package with fewer clients. We could use the v1alpha package to avoid needing all the clients that ultimately aren't used by the test